fuzz-tests/ntt.cpp

```diff
@@ -21,49 +21,52 @@ ll modpow(ll a, ll e) {
 }
 
 typedef vector<ll> vl;
-void ntt(ll* x, ll* temp, ll* roots, int N, int skip) {
-	if (N == 1) return;
-	int n2 = N/2;
-	ntt(x     , temp, roots, n2, skip*2);
-	ntt(x+skip, temp, roots, n2, skip*2);
-	rep(i,0,N) temp[i] = x[i*skip];
-	rep(i,0,n2) {
-		ll s = temp[2*i], t = temp[2*i+1] * roots[skip*i];
-		x[skip*i] = (s + t) % mod; x[skip*(i+n2)] = (s - t) % mod;
-	}
-}
-void ntt(vl& x, bool inv = false) {
-	ll e = modpow(root, (mod-1) / sz(x));
-	if (inv) e = modpow(e, mod-2);
-	vl roots(sz(x), 1), temp = roots;
-	rep(i,1,sz(x)) roots[i] = roots[i-1] * e % mod;
-	ntt(&x[0], &temp[0], &roots[0], sz(x), 1);
+int mul(int a, int b) { return (ll)a * b % mod; }
```

Comment:
You get a -Wconversion warning here, and in a bunch of other places. -Wconversion is used by KACTL's default compile macro, and super important for not getting overflows in number theory-heavy code, so please make sure the code compiles without such warnings. Possibly by using `ll` for all modded numbers (`int` and `ll` should be about as fast on 64-bit systems, and even on 32-bit ones I think perf might be dominated by modulo operations? So with some luck this won't slow anything down. But please do benchmark...)